Feature/student view and answer

theryanfo opened the pull request:
## Pull Request Template

**Referenced Issue:** Student view and answer questions

**Reviewers (if applicable):** 

> [!NOTE]
> Inform your reviewers using Slack on PRs in addition to tagging them on the PR.

**Description of changes**

app/active session created. Added app/api routes. Added components to components/ folder. 

> [!IMPORTANT]
> Run a local build first to ensure everything is working

j3rrythomas commented on app/active-session/[course-session-id]/live-poll/page.tsx:
```diff
@@ -0,0 +1,255 @@
+            try {
+                // Wait for the initial fetch to complete
+                await fetchActiveQuestion();
+
```
seems like the fetchCourseSession is called twice instead of once every interval

theryanfo commented:
## Pull Request Template

**Referenced Issue:** Student view and answer questions

**Reviewers (if applicable):** 

> [!NOTE]
> Inform your reviewers using Slack on PRs in addition to tagging them on the PR.

**Description of changes**

- Created Student view to select course from dashboard and be redirected to active polling session page. 
- There is also an intermediary page for if no active course session is found. The displayed question is determined by the activeQuestionId which is polled for every 5 seconds.
- Added app/api routes. Added components to components/ folder. Added to services/ folder to get CourseSession and Question info
- Pulled from main and resolved conflicts

> [!IMPORTANT]
> Run a local build first to ensure everything is working
